src/govuk/components/header/_index.scss

```diff
@@ -210,6 +210,22 @@
     @include govuk-media-query ($from: tablet) {
       top: govuk-spacing(3);
     }
+
+    @include govuk-media-query ($from: desktop) {
+      display: none;
+    }
+
+    &[hidden] {
+      display: none;
+    }
+
+    .js-enabled & {
+      display: block;
+
+      @include govuk-media-query ($from: desktop) {
+        display: none;
+      }
+    }
```

Comment:
I'm a little bit lost with all of the `display: none` and `display: block` rules and how they interact with the `hidden` attribute.

It feels like we should be able to simplify things quite a lot as we should be able to rely on the hidden attribute? The only exception I think being where the service has not added the hidden attribute to the HTML and the JS doesn't run?

Reply:
There are 2 reasons that we've got any `display: none`'s at all in the CSS:

1. To account for older browser which don't support `hidden` (-IE11)
2. To account for when teams haven't added the initial `hidden` and we don't want to introduce a regression on their no js version of the header ie: the menu button being visible with CSS but without JS.

Do you think these are overcautious? I'm gonna see if there's a way we can slim this down anyway, maybe relying on the initial `display: block`'s without the styling attached to the `hidden`'s.

Reply:
The `display: none`s are probably fine, but I think we can get rid of at least some of the `display: block` rules? The element should 'naturally' be `display: block` as long as the `hidden` attribute isn't present and the media query doesn't match.

Reply:
I believe I've managed to slim this down (not pushed yet) however I think I've discovered why it initially seemed confusing. The awkwardness comes from the fact that services teams might not have updated their markup to include the `hidden` attribute on the menu by default, meaning that if we push this out without accounting for it in CSS then we could introduce a regression. This creates an awkward dynamic where we can reliably use `&[hidden]` to handle display in CSS for the nav list as we're only ever adding it via js but we can't do this for the menu button as it relies on `hidden` in the markup to manage default behaviour and there's a small chance that `hidden` wont be there.

This feels dodgy to me as it means we have 2 different approaches to fallback hiding in the same sass file. Some options:

1. Leave it as is and create an issue to use the `&[hidden]` rule for the menu button as well as the nav list
2. Make it consistent but save this for a breaking change
3. Make it consistent and just ship it

My instinct is 2 but I'm also not sure how serious an issue this if we went with 3. Interested to get some alternative perspectives.

Reply:
Doesn't the JS add the hidden attribute to the menu?

Reply:
To the menu, yes. In theory the menu never needs the hidden attribute without js and it's far less risk if it's visible compared to if an unusable menu button is visible so we can reliably use only `hidden` and fallback CSS to manage it's visibility.

Reply:
Following changes to this in another thread, I feel like the CSS is tidy enough to go back into review. I leave it to the review to decide if this needs further work or not. We can always return to this and fix it in later releases when we're more confident that teams will have added the `hidden` attribute to their mobile menu buttons.